helper/iofs/iofs_test.go

```diff
@@ -0,0 +1,152 @@
+package iofs
+
+import (
+	"errors"
+	"io/fs"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"testing/fstest"
+
+	billyfs "github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/memfs"
+)
+
+type errorList interface {
+	Unwrap() []error
+}
+
+type wrappedError interface {
+	Unwrap() error
+}
+
+// TestWithFSTest leverages the packaged Go fstest package, which seems comprehensive
+func TestWithFSTest(t *testing.T) {
+	t.Parallel()
+	memfs := memfs.New()
+	iofs := Wrap(memfs)
+
+	files := map[string]string{
+		"foo.txt":     "hello, world",
+		"bar.txt":     "goodbye, world",
+		"dir/baz.txt": "こんにちわ, world",
```

Comment:
I would be good that tests take into account OS specific things, such as path separators:

```suggestion
		filepath.Join("dir","baz.txt"): "こんにちわ, world",
```

Reply:
It turns out that `fstest.testFS` isn't written in a Windows-aware way (e.g. https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/testing/fstest/testfs.go;l=147), so I skipped this test on Windows for now.

Reply:
This makes me sad, but not sad enough to re-implement fstest.testFS in this file, given that we have Linux coverage and billy coverage.